StellaNowSDK/src/Services/StellaNowSdk.cs

```diff
@@ -63,9 +63,27 @@ public async Task StartAsync()
         await _connectionStrategy.StartAsync();
     }
 
-    public async Task StopAsync()
+    public async Task StopAsync(bool waitForEmptyQueue = false, TimeSpan? timeout = null)
     {
         _logger?.LogInformation("Stopping");
+        
+        if (waitForEmptyQueue)
+        {
+            var cancellationTokenSource = new CancellationTokenSource(timeout ?? TimeSpan.FromMinutes(1)); // default to 1 minute timeout
+
+            _logger?.LogInformation("Waiting for message queue to empty");
```

Comment:
maybe let's log `MessagesPendingForDispatchCount()` so if this is running long time we see on each loop if there is any change in size